analyzers/tests/SonarAnalyzer.Test/Rules/UnusedPrivateMemberTest.Constructors.cs

```diff
@@ -141,6 +141,25 @@ public interface IInterface
 }
 ").Verify();
 
+        [DataTestMethod]
+        [DataRow("private", "Remove the unused private constructor 'Foo'.")]
+        [DataRow("protected", "Remove unused constructor of private type 'Foo'.")]
+        [DataRow("internal", "Remove unused constructor of private type 'Foo'.")]
+        [DataRow("public", "Remove unused constructor of private type 'Foo'.")]
+        public void UnusedPrivateMember_NonPrivateConstructorInPrivateClass(string accessModifier, string expectedMessage) =>
+            builder.AddSnippet($$$"""
+public class Some
+{
+    private class Foo // Noncompliant
+    {
+        {{{accessModifier}}} Foo() // Noncompliant {{{{{expectedMessage}}}}}
+        {
+            var a = 1;
+        }
+    }
+}
+""").Verify();
```

Comment:
The indentation can be improved.
```suggestion
        public void UnusedPrivateMember_NonPrivateConstructorInPrivateClass(string accessModifier, string expectedMessage) =>
            builder.AddSnippet($$$"""
                public class Some
                {
                    private class Foo // Noncompliant
                    {
                        {{{accessModifier}}} Foo() // Noncompliant {{{{{expectedMessage}}}}}
                        {
                            var a = 1;
                        }
                    }
                }
                """).Verify();
```

Reply:
There are three options:
1. Leave as is
2. Refactor everything
3. Clean as you code

I'd go with 3, but if you think it's too much of a sore on the eye, sticking with 1 is also fine.